src/context/context.ts

```diff
@@ -8,40 +9,41 @@ import {
 } from '@hyperlane-xyz/sdk';
 import { isNullish } from '@hyperlane-xyz/utils';
 
-import { getChainConfigs } from './chains';
-import { getWarpCoreConfig } from './tokens';
+import { assembleChainMetadata } from './chains';
+import { assembleWarpCoreConfig } from './warpCoreConfig';
 
 export interface WarpContext {
-  chains: ChainMap<ChainMetadata & { mailbox?: Address }>;
-  multiProvider: MultiProtocolProvider<{ mailbox?: Address }>;
+  registry: IRegistry;
+  chains: ChainMap<ChainMetadata>;
+  multiProvider: MultiProtocolProvider;
   warpCore: WarpCore;
 }
 
+// Note: This was initially static so it was simpler to keep it out of a state store.
+// Now it's somewhat dynamic based on env vars. If more flexibility is needed, it could be moved.
 let warpContext: WarpContext;
```

Comment:
im not super familiar with this react query context pattern but this does feel like a code smell

Reply:
This has nothing to do with react-query. Though you're right it is a smell... now. It wasn't when the context was simple and static. Of course I considered moving it to a state store (hence the comment) but that would mean going through store access hooks and passing the context around through function params everywhere it's used. Not hard, but a verbose and far-reaching refactor. I accepted this tradeoff